### PR TITLE
Fix description of inst interators in opt::Function

### PR DESCRIPTION
**Pull request by dneto0**

### CODE REVIEW

File: source/opt/function.h [hunk 1]
```diff
@@ -104,19 +104,26 @@ class Function {
     });
   }
 
-  // Runs the given function |f| on each instruction in this function, and
-  // optionally on debug line instructions that might precede them.
+  // Runs the given function |f| on instructions in this function, in order,
+  // and optionally on debug line instructions that might precede them.
+  // If |f| returns false, iteration is terminated.
```
Comment (s-perron) [inline source/opt/function.h]: This is not true for the "ForEachInst" version.  |f| has no return value.

File: source/opt/function.h [hunk 2]
```diff
@@ -104,19 +104,26 @@ class Function {
...
   bool WhileEachInst(const std::function<bool(Instruction*)>& f,
                      bool run_on_debug_line_insts = false);
   bool WhileEachInst(const std::function<bool(const Instruction*)>& f,
                      bool run_on_debug_line_insts = false) const;
 
   // Runs the given function |f| on each parameter instruction in this function,
-  // and optionally on debug line instructions that might precede them.
+  // in order, and optionally on debug line instructions that might precede
+  // them. If |f| returns false, iteration is terminated and this function
+  // returns false.
```
Comment (s-perron) [inline source/opt/function.h]: |f| is a void function, and does not return true or false.

**Comment by s-perron** [comment]
@dneto0 Will you be updating this?

**Comment by dneto0** [comment]
whoops.  yeah, I'll fix. thanks for the review. ;-)

File: source/opt/function.h (hunk 1 above)
Comment (dneto0) [reply source/opt/function.h]: Fixed in next commit

File: source/opt/function.h (hunk 2 above)
Comment (dneto0) [reply source/opt/function.h]: Fixed in next commit.

**Comment by dneto0** [comment]
Hey @s-perron, this is ready for review again

**Pull request merged**